File: DataSets/DataSet-Old/LeetCodeTasks - Testing/Good/69_sqrtx/solutions/java/solution.java
class Solution {
    public int mySqrt(int x) {
        if (x == 0) return 0;
        long left = 1, right = x, mid, sq;
        while (left <= right) {
            mid = left + (right - left) / 2;
            sq = mid * mid;
            if (sq == x) return (int) mid;
            else if (sq < x) left = mid + 1;
            else right = mid - 1;
        }
        return (int) right;
    }
}

File: DataSets/DataSet-Old/LeetCodeTasks - Testing/Good/69_sqrtx/solutions/python3/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0
        left, right = 1, x
        while left <= right:
            mid = (left + right) // 2
            if mid * mid <= x < (mid + 1) * (mid + 1):
                return mid
            elif mid * mid < x:
                left = mid + 1
            else:
                right = mid - 1
